controllers/utils/utils.go

```diff
@@ -27,3 +28,24 @@ func UpdateConditions(existingConditions []metav1.Condition, newCondition metav1
 
 	return append([]metav1.Condition{newCondition}, existingConditions...)
 }
+
+const (
+	letterBytes  = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	specialBytes = "!@#$%^&*()_+-=[]{}\\|;':\",.<>/?`~"
+	numBytes     = "0123456789"
+)
+
+// func GeneratePassword(length int, useLetters bool, useSpecial bool, useNum bool) string {
+func GeneratePassword(length int, useLetters bool, useSpecial bool, useNum bool) string {
+	b := make([]byte, length)
+	for i := range b {
+		if useLetters {
+			b[i] = letterBytes[rand.Intn(len(letterBytes))]
+		} else if useSpecial {
+			b[i] = specialBytes[rand.Intn(len(specialBytes))]
+		} else if useNum {
+			b[i] = numBytes[rand.Intn(len(numBytes))]
+		}
+	}
+	return string(b)
```

Comment:
non-blocking:

> Package rand implements pseudo-random number generators suitable for tasks such as simulation, but it should not be used for security-sensitive work. 
>
> ref: https://pkg.go.dev/math/rand

However, I don't know a secure way either.

Reply:
Maybe  https://pkg.go.dev/crypto/rand is a suitable replacement for https://pkg.go.dev/math/rand

Reply:
implemented with crypto/rand